### PR TITLE
Update knb to recognize drop-in module replace

### PR DESCRIPTION
**Pull request by dsimansk**
## Description

Update `knb` to recognize drop-in replacement for a certain plugin. Update readme. And make sure everything is updated with `hack/update-deps.sh`.

/cc @vyasgun @rhuss 

## Changes

* Update knb to recognize drop-in module replace

## Reference

Fixes #

**Release Note**

```release-note
NONE
```

### CODE REVIEW

File: tools/knb/pkg/plugin/distro.go [hunk 1]
```diff
@@ -131,6 +148,17 @@ func processModuleReplace(plugin Plugin) error {
 	return nil
 }
 
+// goModTidy executes `go mod tidy` command
+func goModTidy() error {
+	//nolint:gosec // Expected go cmd.
+	_, err := exec.Command("go", "mod", "tidy").Output()
+	if err != nil {
+		return fmt.Errorf("go tidy failed: %w", err)
```
Comment (rhuss) [inline tools/knb/pkg/plugin/distro.go:156]:
```suggestion
		return fmt.Errorf("go mod tidy failed: %w", err)
```

**Review by rhuss — APPROVED** [review]
looks good to me, minor nit.

/approve
/lgtm
/hold

**Comment by dsimansk** [comment]
I've merged the suggestion. @rhuss or @vyasgun if you may lgtm again, thanks!

/unhold

**Pull request merged**